column_mapping.go

```diff
@@ -0,0 +1,104 @@
+package parquet
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ColumnMapping is an interface representing types which map column paths to
+// their zero-based index in a parquet schema.
+//
+// ColumnMapping is a read-only type, and therefore can be used concurrently
+// from multiple goroutines.
+type ColumnMapping struct {
+	mapping columnMapping
+	columns [][]string
+}
+
+// ColumnIndex returns the column index of the column at the given path.
+//
+// If the path was not found in the mapping, or if it did not represent a
+// leaf column of the parquet schema, the method returns a negative value.
+func (m *ColumnMapping) ColumnIndex(path []string) (columnIndex int) {
```

Comment:
I'm a little confused about what would constitute a path in this example - normally I'd think of a path as a single string.

Reply:
It represents the sequence of column names that identify a leaf column in the parquet schema. We use `[]string` instead of a separator because all UTF-8 code points are valid in columns names, electing an arbitrary character as separator for a path representation would diverge from the specification and potentially conflict with column names found in parquet files that contained the separator.